File: frontend/bibliotecario.ts
import { Usuario } from './usuario';
import { ClienteFactory } from './user-factory';

export class Bibliotecario extends Usuario {
    cadastrarUsuario(cpf: string, nome: string, credenciais: { usuario: string, senha: string }) {
        return ClienteFactory.criarUsuario(cpf, nome, credenciais);
    }
}

File: src/biblioteca/Bibliotecario.java
package biblioteca;

public class Bibliotecario extends Usuario {
    public Bibliotecario(String cpf, String nome, Credenciais credenciais) {
        super(cpf, nome, credenciais);
    }

    public Cliente cadastrarUsuario(String cpf, String nome, Credenciais credenciais) {
        return ClienteFactory.criarUsuario(cpf, nome, credenciais);
    }
}
